pkg/deviceclaimingserver/enddevices/ttjs/ttjs.go

```diff
@@ -119,51 +118,19 @@ var (
 	errDeviceAccessDenied   = errors.DefinePermissionDenied("device_access_denied", "access to device with `{dev_eui}` denied. Either device is already claimed or owner token is invalid")
 	errUnauthorized         = errors.DefineUnauthenticated("unauthorized", "client API Key missing or invalid")
 	errNoHomeNSID           = errors.DefineInvalidArgument("no_home_ns_id", "no HomeNSID configured for network server address `{address}`")
-	errParseQRCode          = errors.Define("parse_qr_code", "parse QR code failed")
-	errQRCodeData           = errors.DefineInvalidArgument("qr_code_data", "invalid QR code data")
-	errNoJoinEUI            = errors.DefineInvalidArgument("no_join_eui", "failed to extract JoinEUI from request")
 )
 
 // Claim implements EndDeviceClaimer.
-func (client *TTJS) Claim(ctx context.Context, req *ttnpb.ClaimEndDeviceRequest) (*ttnpb.EndDeviceIdentifiers, error) {
+func (client *TTJS) Claim(ctx context.Context, joinEUI *types.EUI64, devEUI *types.EUI64, claimAuthenticationCode string, networkServerAddress string) error {
 	htenantID := client.config.TenantID
 
-	var (
-		joinEUI, devEUI        types.EUI64
-		hNSAddress, ownerToken string
-	)
-	if authenticatedIDs := req.GetAuthenticatedIdentifiers(); authenticatedIDs != nil {
-		joinEUI = authenticatedIDs.JoinEui
-		devEUI = authenticatedIDs.DevEui
-		ownerToken = authenticatedIDs.AuthenticationCode
-	} else if qrCode := req.GetQrCode(); qrCode != nil {
-		data, err := qrcode.Parse(qrCode)
-		if err != nil {
-			return nil, errParseQRCode.WithCause(err)
-		}
-		authIDs, ok := data.(qrcode.AuthenticatedEndDeviceIdentifiers)
-		if !ok {
-			return nil, errQRCodeData.New()
-		}
-		joinEUI, devEUI, ownerToken = authIDs.AuthenticatedEndDeviceIdentifiers()
-	} else {
-		return nil, errNoJoinEUI.New()
-	}
-
-	hNSAddress, _, err := net.SplitHostPort(req.TargetNetworkServerAddress)
-	if err != nil {
-		// TargetNetworkServerAddress is already validated by the API.
-		// An error here means that it does not contain a port, so we use it directly.
-		hNSAddress = req.TargetNetworkServerAddress
-	}
-
-	hNSID, ok := client.hsNSIDs[hNSAddress]
+	hNSID, ok := client.hsNSIDs[networkServerAddress]
```

Comment:
Don't we have to `SplitHostPort` anymore?

Reply:
It's already done at the entry in `grpc_end_devices.go`